src/controllers/app_controller_test.go

```diff
@@ -0,0 +1,238 @@
+package controllers
+
+import (
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"golang-bootcamp-2020/domain/model"
+	_errors "golang-bootcamp-2020/utils/error"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type appServiceMock struct {
+	mock.Mock
+}
+
+type restError struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+var (
+	characters = []model.Character{
+		{
+			Id:       1,
+			Name:     "Rick Sanchez",
+			Status:   "Alive",
+			Species:  "Human",
+			Type:     "",
+			Gender:   "Male",
+			Origin:   model.Nested{Name: "Earth (C-137)", Url: "https://rickandmortyapi.com/api/location/1"},
```

Comment:
By convention, if you are using a hardcoded value more than 2 times, declared as a constant